### PR TITLE
Update bellmans-gapc to source from upstream repo

### PR DESCRIPTION
**Pull request by sjanssen2**
I am now the official maintainer of the Bellman's GAP compiler and language. I migrated the code to the publish git repo https://github.com/jlab/gapc/ added automatic testing via Travis and started to integrate changes in the source code. Currently, this effects aspect of the basic RNA library and fixing come compatibility issues with different Bison versions.

This PR is to redirect sources to the new upstream URL and take advance of smaller code changes mentioned above.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by sjanssen2** [comment]
Removed this recipe from black-list as it should build with various current boost versions, e.g. 1.58 https://launchpadlibrarian.net/461039173/buildlog_ubuntu-xenial-amd64.bellmansgapc_2020.01.08-0~202001171328~ubuntu16.04.1_BUILDING.txt.gz or 1.67: https://launchpadlibrarian.net/461039160/buildlog_ubuntu-disco-amd64.bellmansgapc_2020.01.08-0~202001171328~ubuntu19.04.1_BUILDING.txt.gz

**Comment by sjanssen2** [comment]
I am not sure what causes a `RuntimeError: Couldn't extract raw recipe text for bellmans-gapc output` error. I found https://github.com/conda/conda-build/issues/3860 and therefore try playing with the meta.yaml layout

**Comment by sjanssen2** [comment]
@dpryan79 I could need some help here. I have the feeling my test is passing in principle. How can we solve the `ERRO Task processing failed: API error (500): {"message":"Get https://quay.io/v2/dpryan79/mulled_container/manifests/latest: error parsing HTTP 429 response body: invalid character 'T' looking for beginning of value: \"Too many login attempts. \\nPlease reset your Quay password and try again.\""}
 
08:09:55 BIOCONDA INFO (ERR) 
 ` issue?

**Comment by sjanssen2** [comment]
@BiocondaBot rebuild

**Comment by sjanssen2** [comment]
error message now changed to `11:45:40 BIOCONDA INFO (ERR) [May 26 11:45:40] ERRO Task processing failed: API error (500): {"message":"Get https://quay.io/v2/dpryan79/mulled_container/manifests/latest: unauthorized: Invalid Username or Password"}
 
11:45:40 BIOCONDA INFO (ERR) 
 `

**Comment by sjanssen2** [comment]
@BiocondaBot rebuild

**Comment by sjanssen2** [comment]
@BiocondaBot rebuild

**Comment by sjanssen2** [comment]
I am providing a source file for the test section. It works for the "normal" build, but it seems to be missing for the mulled-build. What is the correct way to pass this file to the mulled build system?
Or should I alternatively remove this functional test? Could need some advice here @dpryan79 @johanneskoester

**Comment by sjanssen2** [comment]
@BiocondaBot please add label

**Comment by sjanssen2** [comment]
Current tests are passing, however I do have a question regarding additional tests, which I commented our for a moment @bioconda/core
They work for the normal build test, however when the mulled build is tested, the test source file seems to be missing. What is the correct way to pass this file into the mulled build/test container?

**Comment by dlaehnemann** [comment]
Regarding the more complicated test that you are suggesting, I think this is not necessary. A basic test checking that the command is installed and executes with an exit status of 0 should be enough. I remember, that getting the respective files copied to the right place can be quite a hassle, and as the contribution guidelines state, this also adds extra storage and testing time requirements:
https://bioconda.github.io/contributor/guidelines.html#tests

So I think if you remove the respective file and the commented out test code, this recipe update should be good to go.

One additional thing you might want to consider: For versioning, [semantic versioning](https://semver.org/) is often assumed and would also play nicer with conda's mechanism to restrict versions to install or as dependencies with operators such as `>=` et al. See e.g. the conda manual here for an example how non-semantic versioning may complicate things:
https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html?highlight=semantic%20versioning#pinning-at-the-recipe-level

Basically, this would just mean moving towards release tags on GitHub that follow the semantic versioning scheme of `MAJOR.MINOR.PATCH`.

**Comment by sjanssen2** [comment]
Hey @dlaehnemann many thanks for your review and clear suggestions. I removed the test source file and the according commands from meta.yaml.

Regarding the version number. I am also unhappy with the way it is done in Bellman's GAP. However, it comes with its own history and is used for e.g. Ubuntu packages https://launchpad.net/~janssenlab/+archive/ubuntu/software which I don't want to change or cannot change backwards. 
In principle, the conda package version number must not be in sync with the program version number (thus we could follow `MAJOR.MINOR.PATCH`), but it might come with it's own downstream issues :-/
I don't see a perfect solution here and would be happy for further opinions.

**Comment by dlaehnemann** [comment]
I see, I guess it is then the cleanest way to keep version numbers in sync across different distribution channels. There are other packages, that work with similar versioning (see e.g. the [art read simulator](https://bioconda.github.io/recipes/art/README.html#recipe-Recipe%20&#x27;art&#x27;)). The only downside is, that conda or the users might confuse what the newest version is. But having separate versioning systems may be just as confusing. So once the tests pass, I'll add a formal review approving the recipe.

File: recipes/bellmans-gapc/meta.yaml [hunk 1]
```diff
@@ -25,9 +23,17 @@ requirements:
...
     - gsl
     - openblas
-source:
-  sha256: a78c39660ebcc7175466c7848f95d33aba277709352a65b6a5c7a2deb5cd10d5
-  url: https://github.com/mmiladi/gapc/archive/git-0.1.tar.gz
+
 test:
+  files:
+    # copy of file in upstream repo: testdata/grammar/nussinov.gap
+    - nussinov.gap
```
Comment (dlaehnemann) [inline recipes/bellmans-gapc/meta.yaml:30]: Can you also remove this, as it is not being used in the tests?

File: recipes/bellmans-gapc/nussinov.gap [hunk 2]
```diff
@@ -0,0 +1,151 @@
+
+signature Nuss(alphabet, answer) {
+
+  answer nil(void);
+  answer right(answer, alphabet);
+  answer pair(alphabet, answer, alphabet);
+  answer split(answer, answer);
+  choice [answer] h([answer]);
+
```
Comment (dlaehnemann) [inline recipes/bellmans-gapc/nussinov.gap:9]: I would remove this entire file, it is not needed without the extended tests, right?

File: recipes/bellmans-gapc/meta.yaml (hunk 1 above)
Comment (sjanssen2) [reply recipes/bellmans-gapc/meta.yaml]: good catch, I thought I did remove this already with my commit, but I must have done it wrong. Should be fine by now.

File: recipes/bellmans-gapc/nussinov.gap (hunk 2 above)
Comment (sjanssen2) [reply recipes/bellmans-gapc/nussinov.gap]: agreed. File is gone.

**Comment by dlaehnemann** [comment]
@BiocondaBot update

**Comment by dlaehnemann** [comment]
@BiocondaBot rebuild

**Comment by dlaehnemann** [comment]
Sorry, I committed a little improvement and now something seems to have changed in the Circle CI setup, causing the linting to fail. And it seems like I can't trigger the bot command that the failed linting suggests. @sjanssen2 could you try posting a comment with just `@BiocondaBot update`? Hope this is enough and you don't need to pull these changes into your local branch. But if this doesn't help, the full error message was:

```
#!/bin/bash -eo pipefail
.circleci/setup.sh

From ssh://github.com/bioconda/bioconda-recipes
 * [new branch]          master     -> __upstream__1LOhK9Ejy2/master
Your bioconda-recipes CI configuration is out of date.
Please update it to the latest version of the upstream master branch.

Have @BiocondaBot attempt to fix this by creating a comment on your PR:

   @BiocondaBot update

Once the update commit has been created, update your local copy of
your branch:

  git pull


You can also fix this manually, e.g., by running:
  git fetch https://github.com/bioconda/bioconda-recipes.git master
  git merge FETCH_HEAD


Exited with code exit status 1
```

**Comment by sjanssen2** [comment]
@BiocondaBot update
